src/type.c

```diff
@@ -1,3 +1,9 @@
+#define _GNU_SOURCE
+#define GNU_SOURCE
+
+#include <stdio.h>
```

Comment:
It's best to put system-wide includes before your local includes and not the other way around. This is needed because when you use structs or macros from these system-wide includes, they need to be declared before your functions. 

I've already seen this a few times in your code base, so you'll want to fix it in those places as well.